subworkflows/local/minimap2_subworkflow.nf

```diff
@@ -0,0 +1,30 @@
+//
+// Minimap2 subworkflow to index an assembly and map reads to the assembly
+//
+
+include { MINIMAP2_INDEX } from '../../modules/nf-core/minimap2/index'
+include { MINIMAP2_ALIGN } from '../../modules/nf-core/minimap2/align'
+
+workflow MINIMAP2_SUBWORKFLOW {
+    take:
+    assembly
+    reads
+
+    main:
+    ch_versions = Channel.empty()
+
+    // build index
+    MINIMAP2_INDEX(assembly)
+    ch_versions = ch_versions.mix(MINIMAP2_INDEX.out.versions)
+    ch_index = MINIMAP2_INDEX.out.index
+
+    // align reads to index
+    MINIMAP2_ALIGN(reads, ch_index, "true", "false", "true")
```

Comment:
Two questions:
* Do we want to parameterize the "true", "false" stuff?
* Are you sure this is the expected usage of the `MINIMAP2_ALIGN`? The strings are throwing me off as I don't know how Groovy passes them on to the underlying process. But if it works like any other language and passes them as a string (rather than a boolean) "false" is still truthy, which is not what you want.

Reply:
I have further dug into how this is actually working in https://github.com/Arcadia-Science/metagenomics/pull/26 and it's not quite what the above is doing I've discovered. I will need to change this because in this case I do actually want the BAM format and not PAF, whereas I had to take out this behavior entirely in the metagenomics workflow.